sha1-name.c

```diff
@@ -1460,10 +1468,16 @@ int strbuf_check_branch_ref(struct strbuf *sb, const char *name)
  * This is like "get_oid_basic()", except it allows "object ID expressions",
  * notably "xyz^" for "parent of xyz"
  */
-int get_oid(const char *name, struct object_id *oid)
+int get_oid_gently(const char *name, struct object_id *oid, int gently)
 {
 	struct object_context unused;
-	return get_oid_with_context(name, 0, oid, &unused);
+	if (gently)
+		gently = GET_OID_GENTLY;
+	return get_oid_with_context(name, gently, oid, &unused);
```

Comment:
Or put it directly into the call:

```c
return get_oid_with_context(name, gently ? GET_OID_GENTLY : 0, oid, &unused);
```

(It a good practice to avoid changing parameters, in particular when you change their role (from a Boolean to a bitfield of flags, in this case)).